routing.go

```diff
@@ -440,28 +440,7 @@ func (dht *IpfsDHT) Provide(ctx context.Context, key cid.Cid, brdcst bool) (err
 		return nil
 	}
 
-	closerCtx := ctx
-	if deadline, ok := ctx.Deadline(); ok {
-		now := time.Now()
-		timeout := deadline.Sub(now)
-
-		if timeout < 0 {
-			// timed out
-			return context.DeadlineExceeded
-		} else if timeout < 10*time.Second {
-			// Reserve 10% for the final put.
-			deadline = deadline.Add(-timeout / 10)
-		} else {
-			// Otherwise, reserve a second (we'll already be
-			// connected so this should be fast).
-			deadline = deadline.Add(-time.Second)
-		}
-		var cancel context.CancelFunc
-		closerCtx, cancel = context.WithDeadline(ctx, deadline)
-		defer cancel()
-	}
-
-	peers, err := dht.GetClosestPeers(closerCtx, string(keyMH))
```

Comment:
I'm all for removing complexity, but are we sure we want to remove this? While our queries should complete now (and hopefully be pretty snappy), if for some reason they don't (or Provide was called with a very short timeout) this might be handy to have around.

Reply:
But does this actually do something useful now? In the past, we continued till we timed out. Now, we continue until we reach the end. Will we ever time out and return a partial result?

Reply:
Note: the annoying part is that we now can't distinguish between "we failed to send this record to anyone" and "we failed to send this record to the right peers".

Reply:
It's a good point, but I'm not sure if we care. If Providing is important to you then you should reattempt at a later time if you had a failure. Without knowing an approximation of how "deep" into the network our search got without timing out (e.g. distance of Kad peers compared to approximated network size) there's also no real information to give about "we sent to the some peers" that's useful.

That being said if you think it's important feel free to add an error that can be checked against. However, I think that's probably overkill.

Reply:
Eh, we can deal with this later if it's an issue.